doc/dart_documentation_comment_specification.md

```diff
@@ -0,0 +1,478 @@
+typedef JsonMap<T> = Map<String, T>;
+
+/// A record typedef.
+///
+/// [a] : Resolves to record field
+typedef MyRecord = ({int a, bool b});
+
+/// A function type typedef.
+///
+/// [p] : Resolves to parameter
+typedef MySecondTypedef<T> = void Function(int p);
+
+/// A function type typedef (old syntax).
+///
+/// [p] : Resolves to parameter
+typedef void MyTypedef<T>(int p);
+
+/// A function type typedef with nested function functions.
+///
+/// [fun] : Resolves to parameter
+/// [p] : is not in scope
+typedef F<T> = void Function(void Function<S>(void Function(T p)) fun);
+```
+
+**Note on Nested Scopes:** The lookup only applies to the *immediate* parameters or fields of the typedef. It does not extend into parameters of nested function types.
+
+### **5.4. Resolving Qualified Names**
+
+When a reference contains a qualified name (e.g., `[prefix.ClassName.member]`), the resolution process is an iterative, left-to-right evaluation of each Identifier.
+
+#### **1\. Resolve the First Identifier**
+
+The tool first resolves the first identifier in the qualified name (e.g., prefix) using the standard lookup process ([Section 5.3](#5.3.-detailed-lookup-process)), starting from the doc comment's current scope.
+
+#### **2\. Resolve Subsequent Identifiers**
+
+Once an identifier is resolved, the element it resolves to establishes a new **"**namespace**"** for resolving the *next* identifier in the chain. This process repeats until the entire qualified name is resolved.
+
+The namespace available depends on the type of element the previous identifier resolved to. Below are the primary cases.
+
+**Namespace-Providing Elements**
+
+If an Identifier resolves to one of the following elements, it establishes a new namespace for resolving the next identifier in the chain.
+
+*Case 1: Import Prefix*
+
+* **Namespace:** The export scope of the imported library, as filtered by any show or hide combinators on the import directive.
+* **Example:** In `[math.pi]`, the identifier `math` resolves to an import prefix (e.g., from `import dart:math' as math;`). The tool then searches the public namespace of dart:math for the identifier pi.
+* **Combinator Example:** If the import was import 'dart:math' as math show sin;, the namespace for math would *only* contain sin. A reference to `[math.pi]` would fail to resolve, as `pi` was not included in the show list.
+
+*Case 2: Class-like top-level declaration*
+
+* **Namespace:** The set of all members declared within that element, including:
+  * Instance Members (fields, methods, etc.)
+  * Static Members
+  * Constructors
+
+* **Notes on Generics:**
+  * The namespace does not include the element's own type parameters. For a class `MyClass<T>`, a reference like `[MyClass.T]` is invalid because T is not a member of `MyClass`'s namespace.
+  * References are also made to the generic declaration, not a specific instantiation (e.g., write `[List.filled]`, not `[List<int>.filled]`).
```

Comment:
Or else? 
Currently it seem like the type argument is ignored, but the linking works.

Reply:
The linking works in dartdoc but not in the IDE. I think we should not support it and it is kind of a bug that we resolve it in dartdoc.